Initiale release of Varmundt's Biosphere dungeons (#7812)
rathena/rathena: npc/re/merchants/biosphere.txt

		next;
		select( "That's a nice place" );
		mes "[Admin beta]";
		mes "But... It looks like there's been a bit of a problem lately. Oh, that's right. It would be nice for an adventurer who is an expert in this field to visit.";
		next;
		mes "[Admin beta]";
		mes "Go to the <NAVI>[gentleman]<INFO>ba_in01,384,38,0,101,0</INFO></NAVI> who manages it, and you will get directions.";
		close;
	}
	end;

OnTouch:
	npctalk "Guest, do you need a guide inside the mansion?", "", bc_self;
	end;
}

// Exchange npcs
ba_in01,358,50,8	script	Omega#ms01	EP17_2_OMEGA_CLEANER,{
	mes "[Omega]";
	mes "Vrr-R-R-R-R-R-R-R-R";
	next;
	mes "[" + strcharinfo(0) + "]";
	mes "It seems to convert the pieces of runes obtained inside the artificial habitat into complete runes.";";
	close2;
	callshop( "barter_biosphere_1" );
	end;

OnWalk1:
	npctalk "Vrr-R-R-R-R-R-R";
	unitstopwalk getnpcid(0), USW_FORCE_STOP;
	unitwalk getnpcid(0),365,50, strnpcinfo(0) + "::OnWalk2";
	end;
OnWalk2:
	unitstopwalk getnpcid(0), USW_FORCE_STOP;
	unitwalk getnpcid(0),358,50, strnpcinfo(0) + "::OnWalk1";
	end;
OnInit:

> 	npcspeed 500;	// todo check

Reviewer: Im not sure but from my aegis i found this
```    NpcMovePoint "오메가#ms01" 900 365 50
    NpcMovePoint "오메가#ms01" 900 358 50```
365/358 and 50 is the location of movement
i guess 900 is the npcspeed, it seems to be correct.

	next;
	mes "[Beta]";
	mes "Then please take a look at my inventory. These beautiful gear...!";
	close2;
	callshop( "barter_biosphere_2" );
	end;

OnTouch:
	npctalk "Guest, do you have any items from the artificial habitat?", "", bc_self;
	end;
}

ba_in01,361,53,5	script	Beta#ms03	4_EP17_SCISSORE,{
	mes "[Beta]";
	mes "Hey, adventurers? That's great. Welcome back. Listen to me. Great equipment that adventurers might be interested in, I know.";
	next;
	mes "[Beta]";
	mes "One day, the rune fragments I brought from the artificial habitat were turned into whole runes by Omega here.";
	next;
	mes "[Beta]";
	mes "I was looking at the runes, and I suddenly remembered something, so I imprinted the runes on the equipment left by the guests using some matching items as a medium. I think this is the genius the guests were talking about.";
	next;
	mes "[Beta]";
	mes "But it's not possible for all equipment, uh, that's only for some equipment... oh wow!! Just take a look. I'll imprint it as a thank you to the customers who help us!";
	close2;
	callshop( "barter_biosphere_3" );
	end;

> }

Reviewer: This NPC have a range 3,3 and missing OnTouch/OnInit
```OnTouch:
	unittalk getnpcid(0, "베타#ms03"), "손님, 기가 막~힌 장비! 내가 알고 있는데!";
	end;
OnInit:
	setunittitle( getnpcid(0), "<일반 장비 각인>");
	end;
}```

ba_in01,363,53,5	script	Beta#ms04	4_EP17_BASKET,3,3,{
	mes "[Beta]";
	mes "Do you have Varmundt's engraving equipment obtained from the artificial habitat?";
	mes "^0000cdIf the equipment is refined at least to 7^000000, you can enchant it.";
	mes "In the case of a ring, it is possible to enchant it at any time.";
	next;
	mes "[Beta]";
	mes "You can enchant up to 2 times if you bring a rune that matches the property of each equipment and a small fee. You can also reset the enchantment.";
	next;
	.@base_index = 52;	// Grassland Enchantment index
	.@s = select( "Grassland Equipment Enchantment", "Fire Equipment Enchantment", "Ice Equipment Enchantment", "Death Equipment Enchantment", "Soul Equipment Enchantment", "Poison Equipment Enchantment", "Church Equipment Enchantment" ) - 1;
	mes "[Beta]";
	mes "Okay, I'll help you enchant right away.";
	close2;
	item_enchant( .@base_index + .@s );
	end;

OnTouch:
	npctalk "If you have Varmundt's engraving equipment, come this way.", "", bc_self;
	end;

> }

Reviewer: Also there is a missing OnInit
```
OnInit:
	setunittitle( getnpcid(0), "<장비 인챈트>");
	end;
```

				close;
			}
			mes "[Admin beta]";
			mes "With no data, we are not strong enough to deal with powerful monsters, so we closed the door and waited for the guests.";
			next;
			select( "Me?" );
			mes "[Admin beta]";
			mes "From what I've observed, the guest likes adventure. I've also seen " + ( Sex ? "him" : "her" ) + " ignite a sense of challenge toward new things in this mansion.";
			mes "The customer is very strong and the response to all variables was very quick and accurate.";
			next;
			mes "[Admin beta]";
			mes "That's why we decided to first open up an artificial habitat that no one has visited in years, to our experienced adventurer guests.";
			next;
			mes "[Admin beta]";
			mes "We want to focus on man-made habitat management while our guests are adventurous with monsters.";
			next;
			select( "It seems to be passing it on somehow..." );
			mes "[Admin beta]";
			mes "Shouldn't you leave the job to the experts? Let's leave it to the experts who are a bit more specialized in dealing with monsters. It is our judgment that there is no one other than the customer who can handle this difficult task.";
			next;
			mes "[Admin beta]";

> 			mes "Besides, not all artificial habitats are open, but only the most urgent cold, flame, grassland, and death specimen environments, so the burden will be small.";

Reviewer 2: ```suggestion
			mes "Besides, not all artificial habitats are open, but only the most urgent ice, fire, grass, and death specimen environments, so the burden will be small.";
```

			next;
			mes "[Admin beta]";
			mes "You may own anything you got from the artificial habitat. One more here.";
			next;
			mes "[Admin beta]";
			mes "Rare items obtained from monsters will be made into strong equipment <NAVI>[here]<INFO>ba_in01,362,48,0,101,0</INFO></NAVI>, so it would be a good idea to take a look at them as adventurers. It's a mutual aid to the last degree.";
			close;
		case 2:

> 			switch( select( "Sample environment - ^0000cdIce^000000", "Sample environment - ^ff0000Fire^000000", "Sample environment - ^006600Grassland^000000", "Sample environment - ^660099Death^000000", "Equipment Combination" ) ) {

Reviewer 2: ```suggestion
			switch( select( "Sample environment - ^0000cdIce^000000", "Sample environment - ^ff0000Fire^000000", "Sample environment - ^006600Grass^000000", "Sample environment - ^660099Death^000000", "Equipment Combination" ) ) {
```

	next;
	mes "[Beta]";
	mes "But it's not possible for all equipment, uh, that's only for some equipment... oh wow!! Just take a look. I'll imprint it as a thank you to the customers who help us!";
	close2;
	callshop( "barter_biosphere_3" );
	end;

OnTouch:
	npctalk "Guest, amazing equipment! I know it!", "", bc_self;
	end;

OnInit:
	setunittitle( getnpcid(0), "<General Equipment Imprinting>" );
	end;
}

// Enchant of basic equipment
ba_in01,363,53,5	script	Beta#ms04	4_EP17_BASKET,3,3,{
	mes "[Beta]";
	mes "Do you have Varmundt's engraving equipment obtained from the artificial habitat?";
	mes "^0000cdIf the equipment is refined at least to 7^000000, you can enchant it.";
	mes "In the case of a ring, it is possible to enchant it at any time.";
	next;
	mes "[Beta]";
	mes "You can enchant up to 2 times if you bring a rune that matches the property of each equipment and a small fee. You can also reset the enchantment.";
	next;
	setarray .@index[0],16,17,18,19,57,58,59;

> 	.@s = select( "Grassland Equipment Enchantment", "Fire Equipment Enchantment", "Ice Equipment Enchantment", "Death Equipment Enchantment", "Soul Equipment Enchantment", "Poison Equipment Enchantment", "Church Equipment Enchantment" ) - 1;

Reviewer 2: ```suggestion
	.@s = select( "Grassland Equipment Enchantment", "Fire Equipment Enchantment", "Ice Equipment Enchantment", "Death Equipment Enchantment", "Soul Equipment Enchantment", "Poison Equipment Enchantment", "Holy Equipment Enchantment" ) - 1;
```

	mes "[Beta]";
	mes "Okay, I'll help you enchant right away.";
	close2;
	item_enchant( .@index[.@s] );
	end;

OnTouch:
	npctalk "If you have Varmundt's engraving equipment, come this way.", "", bc_self;
	end;

OnInit:
	setunittitle( getnpcid(0), "<Equipment enchantment>" );
	end;
}

// Equipment reform
ba_in01,368,55,5	script	Beta#ms05	MD_BETA_SCISSORE_NG,3,3,{
	mes "[Beta]";
	mes "Do you have Varmundt's engraved equipment? Wouldn't you like to make it more powerful?";
	mes "If you bring the ^0000cdequipment refined at least to 7^000000, I will try my best.";
	mes "The ring does not have to be refined.";
	next;
	mes "[Beta]";
	mes "You can rest assured that the level of refinement, cards, enchantments, and grades of your current equipment will be maintained.";
	mes "However, ^0000cdVarmundt Soul, Deadly Poison, and Temple Ring retain only cards and enchantments^000000.";
	mes "Then, run! Is it?";
	next;

> 	switch( select( "Upgrade Grassland Gear", "Upgrade Fire Gear", "Upgrade Ice Gear", "Upgrade Death Gear", "Upgrade Sooul Gear", "Upgrade Venom Gear", "Upgrade Temple Gear" ) ) {

Reviewer 2: ```suggestion
	switch( select( "Upgrade Grass Gear", "Upgrade Fire Gear", "Upgrade Ice Gear", "Upgrade Death Gear", "Upgrade Soul Gear", "Upgrade Poison Gear", "Upgrade Holy Gear" ) ) {
```

		case 1:
			.@item_id = 101935;	// Varmundt_Temple_Reform
			break;
		case 2:
			.@item_id = 101938;	// Varmundt_T_Reform
			break;
		}
		break;
	}
	mes "[Beta]";
	mes "Okay, I'll help you right away.";
	close2;
	item_reform( .@item_id );
	end;

OnTouch:
	npctalk "Do you want to make Varmundt's engraved equipment more powerful?", "", bc_self;
	end;

OnInit:
	setunittitle( getnpcid(0), "<Varmundt Equipment Modification>" );
	end;
}

// Enchant of reformed equipment
ba_in01,371,55,5	script	Beta#ms06	EP17_2_BETA_ITEMKEEPER,3,3,{
	mes "[Beta]";
	mes "If you have ^0000cd7 refined Varmundt's upgraded equipment^000000, how about enchanting it?";
	mes "I don't even look at the refine rate of Varmundt's ring.";
	next;
	mes "[Beta]";
	mes "Of course you need some ingredients too.";
	mes "Requires runes and essences that match the attributes of each equipment, Barmil tickets, and a small fee.";
	mes "In the case of a ring, you need runes for each attribute, magic power, and a small fee.";
	next;
	mes "[Beta]";
	mes "If you bring everything, we can enchant it up to three times. Reset is also possible.";
	next;
	setarray .@index[0],52,53,54,55,60,61,62;

> 	.@s = select( "Grassland Equipment Enchantment", "Fire Equipment Enchantment", "Ice Equipment Enchantment", "Death Equipment Enchantment", "Soul Equipment Enchantment", "Poison Equipment Enchantment", "Temple Equipment Enchantment" ) - 1 ;

Reviewer 2: ```suggestion
	.@s = select( "Grass Equipment Enchantment", "Fire Equipment Enchantment", "Ice Equipment Enchantment", "Death Equipment Enchantment", "Soul Equipment Enchantment", "Poison Equipment Enchantment", "Holy Equipment Enchantment" ) - 1 ;
```
